go/vt/vtgate/executor.go

```diff
@@ -690,10 +690,7 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 				return nil, err
 			}
 			return e.handleOther(ctx, safeSession, sql, bindVars, dest, keyspaces[0], destTabletType, logStats)
-		} else {
-			return e.handleOther(ctx, safeSession, sql, bindVars, dest, destKeyspace, destTabletType, logStats)
```

Comment:
I don't think this is a good idea. You also need to handle e.g. "SHOW SLAVE STATUS", "SHOW CREATE TABLE" and other things. We do that by issuing a "USE <shard>" statement first which seems to work.

Reply:
Actually I removed this else statement in the second commit (is not needed).
I agree with "SHOW SLAVE STATUS" but a "SHOW CREATE TABLE" needs a DB/KeySpace to be specified.